src/main/java/com/google/step/model/User.java

```diff
@@ -37,4 +37,18 @@ public User(long id, String email, String username, String bio) {
     this.bio = bio;
     this.photoBlobKey = Optional.empty();
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null || !(obj instanceof User)) {
+      return false;
+    }
+    User user = (User) obj;
```

Comment:
In `equals()` methods, it's usually convention to name this `other` for clarity.

```suggestion
    User other = (User) obj;
```